Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -169,5 +162,50 @@ - (nonnull NSArray *)filterResponseInterceptors:(nonnull NSArray *)proposedRespo
     return [NSArray arrayWithArray:responseInterceptors];
 }
 
+- (void)processResponse:(NSURLResponse *)response onThread:(NSThread *)thread
+{
+    __block CDTHTTPInterceptorContext *ctx =
+    [[CDTHTTPInterceptorContext alloc] initWithRequest:[self.request mutableCopy]];
+
+    ctx.response = (NSHTTPURLResponse*)response;
+    for (NSObject<CDTHTTPInterceptor> *obj in self.responseInterceptors) {
+        ctx = [obj interceptResponseInContext:ctx];
+    }
+
+    if (ctx.shouldRetry && self.remainingRetries > 0) {
+        // retry
+        self.remainingRetries--;
+        self.inProgressTask = [self makeRequest];
+        [self.inProgressTask resume];
+    } else {
+        MYOnThread(thread, ^{
+            [self.delegate receivedResponse:response];
+        });
+        self.finished = YES;
+    }
+}
+
+- (void)processError:(NSError *)error onThread:(NSThread *)thread
```

Comment:
As an additional comment here, what does "error" mean if we get to this block of code? Does it mean an HTTP 4xx/5xx error or does it mean something went wrong at a lower level, like a timeout/socket close etc?